### PR TITLE
Support CircleMarker shapes

### PR DESCRIPTION
**Pull request by cosme-benito**
Added support for creating CircleMarker shapes.

### CODE REVIEW

**Comment by cosme-benito** [comment]
Found a bug when removing circle markers with right click. When I toggled edit mode off it triggered a Null Pointer Exception. Looking into it now, added WIP to the title.

**Comment by codeofsumit** [comment]
HI @cosme-benito , thank you for your work on this 👍 
Do you mind rebasing this on the latest `develop`? I've added a testing framework and maybe you could even add some tests to this PR.
I hope to get some time to manually check your addition in the next days

**Comment by cosme-benito** [comment]
Merged with the develop branch again, added support for the toolbar (including SVG) and tried to extend the Marker classes when possible.

I'm working on the cypress tests but I'm having a hard time getting it to work. It seems cypress is trying to click on the map to add the circle markers before it's ready to accept clicks. Hopefully I can add them soon.

**Comment by cosme-benito** [comment]
Basics tests added.

**Comment by cosme-benito** [comment]
Merged with the develop branch again.

**Comment by m1dst** [comment]
Just came across this when I found that my circle markers wouldn’t move.  Seems this might fix it?  If so I might have to replicate for a few other types of markers I’ve created (triangles and stars at the moment).  I might be better just merging in to a simple extended marker with options.

Is this PR in a state where it should be tested/merged?

**Comment by cosme-benito** [comment]
Leaflet PM doesn't support CircleMarkers, that's why it doesn't work. This
code is ready for testing and it should fix your problem for CircleMarkers
in particular. Adapting this PR to other kind of markers shouldn't too
hard, extending the Circle markers classes will do most of the work.

**Comment by codeofsumit** [comment]
Reviewing this PR and adding Circle Marker Support to leaflet.pm is next on my list

File: src/js/Edit/L.PM.Edit.CircleMarker.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    // change state
+    this._enabled = true;
+
+    // enable removal for the marker
+    if (!this.options.preventMarkerRemoval) {
+      this._layer.on('contextmenu', this._removeMarker, this);
+    }
+
+    // // init markers
+    this._createHintMarker();
```
Comment (codeofsumit) [inline src/js/Edit/L.PM.Edit.CircleMarker.js]: It seems overkill to have hint markers for circleMarkers to drag them. Please use 

```js
if (this.options.draggable) {
  this.enableLayerDrag();
}
```

to make circleMarkers draggable

File: src/js/Edit/L.PM.Edit.CircleMarker.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    if (!this.enabled()) {
+      this.enable(options);
+    } else {
+      this.disable();
+    }
+  },
+  enabled() {
+    return this._enabled;
+  },
+  enable(options) {
```
Comment (codeofsumit) [inline src/js/Edit/L.PM.Edit.CircleMarker.js]: Please add draggable and snappable as default options like so:

```js
enable(options = {draggable: true, snappable: true }) { 
  // ...
}
```

**Comment by cosme-benito** [comment]
Thanks for the feedback, I've implemented all your suggestions except snapping in edit mode.

I'm having a hard time having it to work, the marker is getting "stuck" after hitting the first snap point.

**Comment by codeofsumit** [comment]
@cosme-benito just tried it. It's crazy difficult. I'm building it right now and will PR it to your branch.

**Comment by codeofsumit** [comment]
I have created a Pull Request to your branch:
https://github.com/cosme-benito/leaflet.pm/pull/1

File: src/js/Toolbar/L.PM.Toolbar.js [hunk 3]
```diff
@@ -220,6 +222,22 @@ const Toolbar = L.Class.extend({
       actions: ['cancel'],
     };
 
+    const drawCircleMarkerButton = {
+      title: 'Draw Circle Marker',
```
Comment (codeofsumit) [inline src/js/Toolbar/L.PM.Toolbar.js]: Oh another thing, we have recently released a way to internationalize the plugin. Please check the other buttons and add these English titles and tests to the translation file and include it via like you can see in the other buttons. That way, the community can translate the texts into other languages.

**Comment by cosme-benito** [comment]
Thanks a lot for your help, I don't think I could have figured the problem on my own! From my testing everything appears to be working as intended.

I didn't notice the new support for translation files, that's nice. I'll create a pull request for the portuguese language soon. I added the translations for the circle marker texts to every language except russian and romanian. The text will be in english for those languages until someone more apt than me can translate it.

Also added some tests to ensure the labels are properly translated.

**Comment by cosme-benito** [comment]
I added the NL language file that was merged into the dev branch. 

For Dutch, Romanian and Russian languages I just copied the labels for the "Markers" which I feel is better than just writing it in English. Hopefully someone will translate it in a better way.

**Comment by cosme-benito** [comment]
Merged with dev again.

@codeofsumit Am I missing any request change?

**Comment by codeofsumit** [comment]
@cosme-benito No you don't, sorry for not replying sooner. I got married last weekend and I'm now on honeymoon. I'll check this again when I'm back!

**Comment by cosme-benito** [comment]
Congratulations mate!!!

**Comment by m1dst** [comment]
Would be great to see some love on this PR and finally get it into production.

**Comment by codeofsumit** [comment]
I agree. Everything seems to be working @cosme-benito - I will merge it now :-)

Thanks so much for your patience!!! 🙏

**Pull request merged**